Sources/Build/BuildDescription/SwiftTargetBuildDescription.swift

```diff
@@ -546,7 +546,8 @@ public final class SwiftTargetBuildDescription {
         }
 
         result.append("-c")
-        result.append(contentsOf: self.sources.map(\.pathString))
+        // FIXME: Eliminate side effect.
+        try result.append("@\(self.writeFileList(includeDerivedSources: true).pathString)")
```

Comment:
should this be conditional on the number of sources? seems like it may be easier to debug when not using a file so using it only when we _need_ could be a good strategy?

Reply:
It is not really the number of sources that is determinant of the need, it is the length of the command line.  We do have https://github.com/apple/swift-driver/blob/main/Sources/SwiftDriver/Utilities/System.swift#L58-L116 that can help identify if we are exceeding the length.

I only disagree with your statement in the "may be" aspect - I would say it *is* easier to debug with the command line being complete.

Reply:
is detecting the length an API we can use, seems internal to driver

Reply:
@compnerd I am happy to take this change in its current form if it is put behind `#if windows`